ansible/roles/test/files/ptftests/hash_test.py

```diff
@@ -236,14 +234,22 @@ def check_ipv4_route(self, hash_key, src_port, dst_port_list):
                     dport,
                     src_port))
         logging.info('Expect Ether(src={}, dst={})/IP(src={}, dst={})/TCP(sport={}, dport={})'\
-            .format(exp_router_mac,
+            .format('any',
                     'any',
                     ip_src,
                     ip_dst,
                     sport,
                     dport))
 
-        return verify_packet_any_port(self, masked_exp_pkt, dst_port_list)
+        rcvd_port, rcvd_pkt = verify_packet_any_port(self, masked_exp_pkt, dst_port_list)
+        exp_src_mac = self.router_macs[self.ptf_test_port_map[str(dst_port_list[rcvd_port])]['target_dut']]
+        actual_src_mac = Ether(rcvd_pkt).src
+        if exp_src_mac != actual_src_mac:
+            logging.warn(
```

Comment:
Similar concern here, why generate warning rather than fail the test?

Reply:
resolved